hazelcast/src/main/java/com/hazelcast/internal/monitors/PerformanceLogFile.java

```diff
@@ -0,0 +1,322 @@
+    private final BodyRendered bodyRendered = new BodyRendered();
+    private final String pathname;
+    private long lastAccessed;
+    private int index;
+    private BufferedWriter writer;
+    private boolean renderHead;
+    private int maxRollingFileCount;
+    private int maxRollingFileSizeBytes;
+
+    PerformanceLogFile(PerformanceMonitor performanceMonitor) {
+        this.logger = performanceMonitor.logger;
+        this.hazelcastInstance = performanceMonitor.hazelcastInstance;
+        Node node = hazelcastInstance.node;
+        this.metricRegistry = node.nodeEngine.getMetricsRegistry();
+        this.operationService = node.nodeEngine.getOperationService();
+        this.pathname = "performance-" + hazelcastInstance.getName() + "-" + currentTimeMillis() + "-%03d.log";
+        GroupProperties props = node.getGroupProperties();
+        this.maxRollingFileCount = props.PERFORMANCE_MONITOR_MAX_ROLLED_FILE_COUNT.getInteger();
+
+        // we accept a float so it becomes easier to testing to create a small file.
+        this.maxRollingFileSizeBytes = round(ONE_MB * props.PERFORMANCE_MONITOR_MAX_ROLLED_FILE_SIZE.getFloat());
+
+        if (logger.isFinestEnabled()) {
+            logger.finest("max rolling file size: " + maxRollingFileSizeBytes);
+            logger.finest("max rolling file count: " + maxRollingFileCount);
+        }
+    }
+
+    void render() {
+        try {
+            if (logFile == null) {
+                logFile = new File(format(pathname, index));
+                renderHead = true;
+            }
+
+            if (writer == null) {
+                writer = new BufferedWriter(new FileWriter(logFile, true));
+            }
+
+            if (renderHead) {
+                headRenderer.render(writer);
+                renderHead = false;
+            }
+
+            bodyRendered.render(writer);
+            writer.flush();
+            lastAccessed = logFile.lastModified();
+
+            if (logFile.length() > maxRollingFileSizeBytes) {
+                closeResource(writer);
+                writer = null;
+                logFile = null;
+                index++;
+                deleteOld();
+            }
+        } catch (IOException e) {
+            logIOError(e);
+            logFile = null;
+            closeResource(writer);
+            writer = null;
+        }
+    }
+
+    private void logIOError(IOException e) {
+        if (logger.isFinestEnabled()) {
+            logger.finest("PerformanceMonitor failed to output to file:"
+                    + logFile.getAbsolutePath() + " cause:" + e.getMessage(), e);
+        } else {
+            logger.warning("PerformanceMonitor failed to output to file:"
+                    + logFile.getAbsolutePath() + " cause:" + e.getMessage());
+        }
+    }
+
+    private void deleteOld() {
+        File file = new File(format(pathname, index - maxRollingFileCount));
+        file.delete();
+    }
+
+    public boolean forceRender() {
```

Comment:
it's a public method (even just for internal usage), could document it? I'm not quite sure what is it supposed to do.